lib/active_job/queue_adapters/sneakers_adapter.rb

```diff
@@ -0,0 +1,39 @@
+module ActiveJob
+  module QueueAdapters
+    # Explicitly remove the implementation existing in older rails'.
+    remove_const(:SneakersAdapter) if defined?(:SneakersAdapter)
```

Comment:
Is this common pattern?

Reply:
Yes, same pattern is used for [Sidekiq](https://github.com/sidekiq/sidekiq/pull/6430/files#diff-ebb94e42d8a2439ff96291e25860212cec96b8d9372a767a4f673255ea2c15c0R20) and [Resque](https://github.com/resque/resque/pull/1906/files#diff-cb877f1a4bb4a04694b677275f2b9f36d069a64f29bb94be7b9a68c708bd4795R18)  extraction. Zaak explained it [here](https://github.com/resque/resque/pull/1906#issuecomment-2379449281) into details

Reply:
ok, thanks for the info